src/main/java/net/neoforged/neoforge/event/EventHooks.java

```diff
@@ -302,17 +313,31 @@ public static SpawnGroupData onFinalizeSpawn(Mob mob, ServerLevelAccessor level,
     }
 
     /**
-     * Returns the FinalizeSpawn event instance, or null if it was canceled.<br>
-     * This is separate since mob spawners perform special finalizeSpawn handling when NBT data is present, but we still want to fire the event.<br>
-     * This overload is also the only way to pass through a {@link BaseSpawner} instance.
+     * Finalizes the spawn of a mob by firing the {@link FinalizeSpawnEvent} and calling {@link Mob#finalizeSpawn} with the result.
+     * <p>
+     * This method is separate since mob spawners perform special finalizeSpawn handling when NBT data is present, but we still want to fire the event.
```

Comment:
Isn't spawn NBT data gone?

Reply:
The `CompoundTag` that was present in `finalizeSpawn` is gone, but this line isn't referring to that.

This is referring to the fact that spawners do not normally call `finalizeSpawn` if a `SpawnData` object is present that contains more information than just the `id` key.